model/task_history.go

```diff
@@ -688,6 +700,232 @@ func buildTestHistoryQuery(testHistoryParameters *TestHistoryParameters) ([]bson
 	return pipeline, nil
 }
 
+func testHistoryV2Results(params *TestHistoryParameters) ([]task.Task, error) {
+	if params == nil {
+		return nil, errors.New("unable to determine parameters for test history query")
+	}
+	// run just the part of the query on the tasks collection to form our starting result set
+	tasksQuery, err := formQueryFromTasks(params)
+	if err != nil {
+		return nil, err
+	}
+	projection := bson.M{
+		task.DisplayNameKey:         1,
+		task.BuildVariantKey:        1,
+		task.StatusKey:              1,
+		task.RevisionKey:            1,
+		task.IdKey:                  1,
+		task.ExecutionKey:           1,
+		task.RevisionOrderNumberKey: 1,
+		task.OldTaskIdKey:           1,
+		task.StartTimeKey:           1,
+		task.FinishTimeKey:          1,
+		task.ProjectKey:             1,
+		task.DetailsKey:             1,
+	}
+	tasks, err := task.Find(db.Query(tasksQuery).Project(projection))
+	if err != nil {
+		return nil, err
+	}
+	oldTasks, err := task.FindOld(db.Query(tasksQuery).Project(projection))
+	if err != nil {
+		return nil, err
+	}
+	tasks = append(tasks, oldTasks...)
+
+	// to join the test results, merge test results for all the tasks...
+	if len(params.TestNames) == 0 {
+		out := []task.Task{}
+		taskChan := make(chan task.Task, len(tasks))
+		for _, t := range tasks {
+			taskChan <- t
+		}
+		close(taskChan)
+		wg := sync.WaitGroup{}
+		numWorkers := runtime.NumCPU()
```

Comment:
I would suggest setting this to a constant rather than `runtime.NumCPU()`, since the bottleneck is connections to the DB, not the app server, so this number shouldn't increase if we change the size of the app server.

Reply:
changed to 16